astropy/utils/masked/core.py

```diff
@@ -902,7 +902,7 @@ def __array_function__(self, function, types, args, kwargs):
             except NotImplementedError:
                 return self._not_implemented_or_raise(function, types)
 
-            if not isinstance(dispatched_result, tuple):
+            if dispatched_function._direct_results:
```

Comment:
With my main suggestion, this branch is not necessary any more, but I think I still like the ability to just return `None`, so this would be
```
if dispatched_result is None:
    return
```

Reply:
p.s. Note that with `result, None, None`, the fall-through does the right thing, of just returning `result`. I don't quite understand why it wasn't a tuple of 3 to start with, but likely I just started with returning the plain result, and then realized there were many cases where I had to attach the mask or deal with `out`, so might as well have the option to return a tuple of three.